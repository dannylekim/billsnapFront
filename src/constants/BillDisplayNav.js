export const navItems = [
    {
        name  : "allBills",
        title : "All bills"
    },
    {
        name  : "amountOwed",
        title : "Amount owed"
    },
    {
        name  : "amountIndebted",
        title : "Amount indebted"
    },
    {
        name  : "addBill",
        title : "+ Add bill"
    }
]