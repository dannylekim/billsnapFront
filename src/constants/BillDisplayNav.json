[
  {
    "name": "allBills",
    "title": "All bills"
  },
  {
    "name": "owedToYou",
    "title": "Bills owed to you"
  }
]
