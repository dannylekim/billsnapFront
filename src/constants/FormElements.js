export const createRegisterFormElements = () => {
  const firstNameToolTip = document.createElement("div");
  const lastNameToolTip = document.createElement("div");
  const emailToolTip = document.createElement("div");
  const passwordToolTip = document.createElement("div");
  const verifyPasswordToolTip = document.createElement("div");

  firstNameToolTip.setAttribute("id", "firstName");
  document.body.appendChild(firstNameToolTip);

  lastNameToolTip.setAttribute("id", "lastName");
  document.body.appendChild(lastNameToolTip);

  emailToolTip.setAttribute("id", "email");
  document.body.appendChild(emailToolTip);

  passwordToolTip.setAttribute("id", "password");
  document.body.appendChild(passwordToolTip);

  verifyPasswordToolTip.setAttribute("id", "confirmPassword");
  document.body.appendChild(verifyPasswordToolTip);
};
