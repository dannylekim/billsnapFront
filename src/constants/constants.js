/** file dedicated to constants that are used across the application **/

export const BILLSNAP_TOKEN = "billSnap_token";
export const BILLS = "bills";
