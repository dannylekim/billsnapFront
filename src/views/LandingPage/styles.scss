div.page__landing {
  display: -webkit-box;  /* iOS 6-, Safari 3.1-6 */
  display: -ms-flexbox;  /* IE 10 */
  display: -webkit-flex;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 80%;
  height:100%;
  margin-top: 100px;

}

div#flex__item{
  display: flex;
}

div.title__logo{
  font-size: 15px;
}

div.form__inputs{
  width:70%;
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
}

input.register__login__inputs{
  font-size: 15x;
  border: 3px solid #47E5B6;
  border-radius: 50px;
}

input.register__login__inputs:hover {
  border: 3px solid #C0FFEC;
}

input.register__login__inputs:active, input.register__login__inputs:focus, input.register__login__inputs:focus:hover {
  border: 3px solid #3BD0A3;
}

button.login_register__submit__button.btn.btn-primary{
  background: #30E3AE;
  border: 1px solid #47E5B6;
  font-size: 15px;
}

button.login_register__submit__button.btn.btn-primary:hover, button.login_register__submit__button.btn.btn-primary:active,
button.login_register__submit__button.btn.btn-primary:focus:hover,button.login_register__submit__button.btn.btn-primary:active:focus{
  background: #30E3AE;
  border: 1px solid #47E5B6;
}

button.form__toggle.btn.btn-primary, button.form__toggle.btn.btn-primary:hover, button.form__toggle.btn.btn-primary:active:focus {
  background-color: #3BD0A3;
  border-color: #3BD0A3;
  margin-left: 15px;
  box-sizing: border-box;
  border-radius: 50px;
}

div.tooltip__inner{
  text-shadow: 1px 1px black;
  border: 1px solid black;
  background-color: rgb(71, 70, 70);
}

div.hidden__div{
  padding: .75rem 1.25rem;
  margin-bottom: 1rem;
  height:48px;
}

h6 a.form__toggle{
  color:rgba(165, 199, 245, 0.801);
}

div.form__seperator{
  display: flex;
  font-size: 18px;
  text-align: center;
}

hr.form__horizontal__line{
  width: 25%;
  border: 0.1px solid rgba(0, 0, 0, 0.63);
}

img.character__icon__image{
  height:50px;
}