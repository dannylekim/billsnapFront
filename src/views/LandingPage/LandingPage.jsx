import React, {useState} from "react";
import TitleContent from "../../components/TitleContent";
import RegisterForm from "../RegisterForm";
import LoginForm from "../LoginForm";
import Navbar from "../../components/Navbar";
import "./styles.scss";

/**
 * @description container component that should be in the middle of the login page
 */

export default (props) => {
    const [formType, setFormType] = useState("login");
    const {history} = props;
    return (
        <div className="page__landing">
            <Navbar/>
            <div className="page__content">
                <TitleContent/>
                <div>
                    {formType === "register" ? (
                        <RegisterForm setFormType={setFormType} history={history}/>
                    ) : (
                        <LoginForm setFormType={setFormType} history={history}/>
                    )}
                </div>
            </div>
        </div>
    );
};
