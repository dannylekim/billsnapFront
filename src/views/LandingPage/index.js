import LandingPage from './LandingPage.jsx';
import { connect } from 'react-redux'
import { toggleLandingPageForm } from '../../redux/actions/applicationActions';

const mapStateToProps = (state) => ({
    formType: state.application.showRegisterFirst ? 'register' : 'login'
  })
  
const mapDispatchToProps = (dispatch) => ({
    toggleFormType: () => dispatch(toggleLandingPageForm())
})

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);