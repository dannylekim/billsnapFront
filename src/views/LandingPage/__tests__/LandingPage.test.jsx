import React from "react";
import LandingPage from "../../LandingPage";
import {createRegisterFormElements} from "../../../constants/FormElements";
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";

const mockStore = configureStore();

const store = mockStore({
    application: {
        showRegisterFirst: true,
        isBillLoading: false
    }
});

const store2 = mockStore({
    application: {
        showRegisterFirst: false,
        isBillLoading: false
    }
});

describe("LandingPage", () => {
    let mockFn;
    beforeEach(() => {
        mockFn = jest.fn();
    });

    describe("render", () => {
        describe("snapshots ðŸ“¸", () => {
            it("LandingPage should match snap shot register", () => {
                createRegisterFormElements();
                matches(<Provider store={store}>
                    <LandingPage formType={'register'} toggleFormType={mockFn}/>
                </Provider>);
            });
            it("LandingPage should match snap shot login", () => {
                createRegisterFormElements();
                matches(<Provider store={store2}> <LandingPage formType={'login'} toggleFormType={mockFn}/>
                </Provider>);
            });
        });
    });
});

