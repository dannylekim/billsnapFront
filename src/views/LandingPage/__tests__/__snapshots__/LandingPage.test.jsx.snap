// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage render snapshots ðŸ“¸ LandingPage should match snap shot 1`] = `
<div
  className="page__landing"
>
  <div
    className="title__logo"
    id="flex__item"
  >
    <div
      className="title"
    >
      <div
        className="title__row"
      >
        <h1
          className="title__title"
        >
          Billsnap
        </h1>
        <img
          alt="Logo"
          className="title__logo"
          src="https://via.placeholder.com/150x100.png?text=150x100+Logo"
        />
      </div>
      <div
        className="title__row title__text"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Atque dignissimos eos error quo, repellendus repudiandae sunt. Amet asperiores atque commodi, deleniti dicta expedita ipsa modi nemo, quos rem voluptate voluptates.
        </p>
      </div>
    </div>
  </div>
  <div
    id="flex__item"
  >
    <div
      className="register__container"
    >
      <div
        className="hidden__div"
      />
      <img
        alt="character logo"
        className="character__icon__image"
        src="./billSnapIcon.png"
      />
      <form
        className=""
      >
        <div
          className="form__inputs"
        >
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              autoComplete="first-name"
              className="register__login__inputs form-control"
              id="firstName"
              name="firstName"
              placeholder="ðŸ‘¤ Enter your first name"
              type="text"
            />
          </div>
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              autoComplete="last-name"
              className="register__login__inputs form-control"
              id="lastName"
              name="lastName"
              placeholder="ðŸ‘¤ Enter your last name"
              type="text"
            />
          </div>
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              autoComplete="username"
              className="register__login__inputs form-control"
              id="email"
              name="email"
              placeholder="@ Enter your email"
              type="email"
            />
          </div>
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              autoComplete="new-password"
              className="register__login__inputs form-control"
              id="password"
              name="password"
              placeholder="ðŸ”’ Enter a password"
              type="password"
            />
          </div>
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              autoComplete="new-password"
              className="register__login__inputs form-control"
              id="confirmPassword"
              name="confirmPassword"
              placeholder="ðŸ”’ Verify password"
              type="password"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <button
            className="login_register__submit__button btn btn-primary btn-sm btn-pill"
            name="submit"
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
        </div>
      </form>
      <div>
        <div
          className="form__seperator"
        >
          <hr
            className="form__horizontal__line"
          />
          Or
          <hr
            className="form__horizontal__line"
          />
        </div>
        <div>
          <h6>
            Have an account?
          </h6>
          <button
            className="form__toggle btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            ðŸ‘‹ Log in to your account.
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
